src/main/scala/com/redis/RedisClient.scala

```diff
@@ -13,6 +15,16 @@ object RedisClient {
   case object SUM extends Aggregate
   case object MIN extends Aggregate
   case object MAX extends Aggregate
+
+  private def extractDatabaseNumber(connectionUri: java.net.URI) = {
+    Option(connectionUri.getPath).map(path =>
+      if (path.isEmpty) 0
+      else Try(Integer.parseInt(path.tail)) match {
+        case Success(dbNumber) => dbNumber
+        case Failure(e) => throw new IllegalArgumentException("Invalid database number", e)
+      })
+      .getOrElse(0)
+  }
 }
```

Comment:
1. ah .. I forgot it's Java and we can have nulls :-) .. Cool. Let's have the `Option` then instead of doing a premature optimization.
2. Let's remove the `Try` - it doesn't add much value
3. also could u please explicitly mention the return type of the new function that u added ?

Thanks.